### ISSUE TITLE
How to identify the problem? solargraph scan fails with [NoMethodError]: undefined method `empty?' for nil:NilClass

### ISSUE THREAD

**Issue by adis-io** [castwide/solargraph#243]
Whole output:
```
✗ solargraph scan
warning: parser/current is loading parser/ruby23, which recognizes
warning: 2.3.8-compliant syntax, but you are running 2.3.3.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
[INFO] Indexing workspace files in /Users/myuser/myproject
[INFO] Loading gems for bundler/require
[INFO] Using cached documentation for activesupport at /Users/myuser/.solargraph/cache/gems/activesupport-5.0.0.1/yardoc
[INFO] Using cached documentation for railties at /Users/myuser/.solargraph/cache/gems/railties-5.0.0.1/yardoc
[INFO] Using cached documentation for activemodel at /Users/myuser/.solargraph/cache/gems/activemodel-5.0.0.1/yardoc
[INFO] Using cached documentation for activejob at /Users/myuser/.solargraph/cache/gems/activejob-5.0.0.1/yardoc
[INFO] Using cached documentation for activerecord at /Users/myuser/.solargraph/cache/gems/activerecord-5.0.0.1/yardoc
[INFO] Using cached documentation for actionpack at /Users/myuser/.solargraph/cache/gems/actionpack-5.0.0.1/yardoc
[INFO] Using cached documentation for actionmailer at /Users/myuser/.solargraph/cache/gems/actionmailer-5.0.0.1/yardoc
[INFO] Using cached documentation for actionview at /Users/myuser/.solargraph/cache/gems/actionview-5.0.0.1/yardoc
[INFO] Using cached documentation for actioncable at /Users/myuser/.solargraph/cache/gems/actioncable-5.0.0.1/yardoc
[INFO] Using cached documentation for actionpack at /Users/myuser/.solargraph/cache/gems/actionpack-5.0.0.1/yardoc
[INFO] Using cached documentation for activesupport at /Users/myuser/.solargraph/cache/gems/activesupport-5.0.0.1/yardoc
[INFO] Using cached documentation for actionview at /Users/myuser/.solargraph/cache/gems/actionview-5.0.0.1/yardoc
[INFO] Using cached documentation for actionpack at /Users/myuser/.solargraph/cache/gems/actionpack-5.0.0.1/yardoc
[INFO] Using cached documentation for actioncable at /Users/myuser/.solargraph/cache/gems/actioncable-5.0.0.1/yardoc
[INFO] Using cached documentation for activejob at /Users/myuser/.solargraph/cache/gems/activejob-5.0.0.1/yardoc
[INFO] Using cached documentation for actionmailer at /Users/myuser/.solargraph/cache/gems/actionmailer-5.0.0.1/yardoc
[INFO] Using cached documentation for activemodel at /Users/myuser/.solargraph/cache/gems/activemodel-5.0.0.1/yardoc
[INFO] Using cached documentation for railties at /Users/myuser/.solargraph/cache/gems/railties-5.0.0.1/yardoc
[INFO] Using cached documentation for activerecord at /Users/myuser/.solargraph/cache/gems/activerecord-5.0.0.1/yardoc
[INFO] Error parsing complex type: Unclosed subtype in Array<Integer
[INFO] Error parsing complex type: Unclosed subtype in Array<Object
[INFO] Error parsing complex type: Unclosed subtype in Array<String
[INFO] Error parsing complex type: Unclosed subtype in Array<String, nil
[INFO] Error parsing complex type: Unclosed subtype in Array<String, Hash
[INFO] Error parsing complex type: Unclosed subtype in Array<String, Hash
[INFO] Error parsing complex type: Unclosed subtype in Array<String
Error testing ActiveModel::Serializer::Association#name= (/Users/myuser/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/active_model_serializers-0.10.2/lib/active_model/serializer/association.rb 11) at /Users/myuser/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/active_model_serializers-0.10.2/lib/active_model/serializer/association.rb:12
[NoMethodError]: undefined method `empty?' for nil:NilClass
/Users/myuser/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/solargraph-0.37.2/lib/solargraph/pin/base.rb:240:in `parse_comments'
/Users/myuser/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/solargraph-0.37.2/lib/solargraph/pin/base.rb:99:in `docstring'
/Users/myuser/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/solargraph-0.37.2/lib/solargraph/pin/base_method.rb:47:in `generate_complex_type'
/Users/myuser/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/solargraph-0.37.2/lib/solargraph/pin/base_method.rb:18:in `return_type'
/Users/myuser/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/solargraph-0.37.2/lib/solargraph/pin/base.rb:141:in `typify'
/Users/myuser/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/solargraph-0.37.2/lib/solargraph/pin/base_method.rb:26:in `typify'
/Users/myuser/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/solargraph-0.37.2/lib/solargraph/shell.rb:169:in `block (2 levels) in scan'
/Users/myuser/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/solargraph-0.37.2/lib/solargraph/shell.rb:166:in `each'
/Users/myuser/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/solargraph-0.37.2/lib/solargraph/shell.rb:166:in `block in scan'
/Users/myuser/.rbenv/versions/2.3.3/lib/ruby/2.3.0/benchmark.rb:293:in `measure'
/Users/myuser/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/solargraph-0.37.2/lib/solargraph/shell.rb:164:in `scan'
/Users/myuser/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
/Users/myuser/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
/Users/myuser/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
/Users/myuser/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
/Users/myuser/.rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/solargraph-0.37.2/bin/solargraph:5:in `<top (required)>'
/Users/myuser/.rbenv/versions/2.3.3/bin/solargraph:22:in `load'
/Users/myuser/.rbenv/versions/2.3.3/bin/solargraph:22:in `<main>'
```

Removed `active_model_serializers` from Gemfile and it works.
Tried to run scan in this gems directory - it works.
Tried to run in `app/serializers` - works everything fine.

So what could be the problem?

**Comment by castwide** [comment]
The source of the problem is mapping for the `ActiveModel::Serializer::Association#name=` attribute method. When Solargraph tries to parse the associated comments to get the attribute's description, it receives `nil` instead of a `String`.

The error happens when all of the following are true:
1. The attribute was defined with an `@attr` tag.
2. The map was generated from YARD documentation instead of workspace code (e.g., a gem dependency).
3. The attribute's description is blank.

The `@attr` tag is deprecated, but it still shouldn't cause a fatal error. I'll have a fix for it in the next release.

Incidentally, the latest version of `active_model_serializers` (0.10.10) no longer uses the `@attr` tag and does not trigger the error.

**Comment by castwide** [comment]
Version 0.38.0 fixes the problem with processing attributes from `@attr` tags in YARD documentation.

**Comment by castwide** [comment]
This issue should be resolved in 0.38.0. If the problem persists, please let me know.
